pkg/manifest/service.go

```diff
@@ -335,18 +343,36 @@ func (sr ServiceResource) GetConfigMapKey() string {
 	return DEFAULT_RESOURCE_ENV_NAME
 }
 
-// skipcq
 func (s Service) AnnotationsMap() map[string]string {
-	annotations := map[string]string{}
+	annotations := make(map[string]string)
+
+	for _, annotation := range s.Annotations {
+		// Marshal the value to a JSON string
+		jsonValue, err := json.Marshal(annotation.Value)
+		if err != nil {
+			// Handle JSON marshaling errors if needed
+			fmt.Printf("Error marshaling annotation value for key %s: %v\n", annotation.Key, err)
+			continue
+		}
 
-	for _, a := range s.Annotations {
-		parts := strings.SplitN(a, "=", 2)
-		annotations[parts[0]] = parts[1]
+		jsonStr := string(jsonValue)
+
+		if containsJSONStructure(jsonStr) {
+			key := fmt.Sprintf("%s : |", annotation.Key)
+			annotations[key] = jsonStr
+		} else {
+			key := fmt.Sprintf("%s :", annotation.Key)
+			annotations[key] = jsonStr
+		}
```

Comment:
why json marshal and json str here?

Reply:
This is to format the json structure of the annotation properly.